frontend/src/app/chart/chart.component.ts

```diff
@@ -161,70 +184,119 @@ export class ChartComponent implements OnInit, OnDestroy {
+
+    this.svgLine = this.svgChart.append('g');
 
     const setFocus = (container: d3.ContainerElement) => {
       // Recover coordinate of cursor.
       const mouseFocus = this.xScale.invert(d3.mouse(container)[0]);
       if (mouseFocus === undefined) return;
 
       const timeParse = d3.timeParse('%Q');
-      const dateFormat = d3.timeFormat('%Y-%m-%d');
+      const dateFormat = d3.timeFormat('%Y %b %d');
       const timeFormat = d3.timeFormat('%H:%M:%S.%L');
 
-      // Select the record closest to the cursor.
-      const bisectLeft = d3.bisector((d: Record) => d.time).left;
-      const bisectRight = d3.bisector((d: Record) => d.time).right;
       for (const recordsOneChannel of this.records) {
-        const left = bisectLeft(recordsOneChannel.data, mouseFocus);
-        const right = bisectRight(recordsOneChannel.data, mouseFocus);
-
-        if (!recordsOneChannel.data[left] || !recordsOneChannel.data[right])
-          return;
-        const leftRecord = recordsOneChannel.data[left];
-        const rightRecord = recordsOneChannel.data[right];
-
-        const index =
-          Math.abs(leftRecord.time - mouseFocus) >
-          Math.abs(rightRecord.time - mouseFocus)
-            ? right
-            : left;
-        const selectedData = recordsOneChannel.data[index];
+        if (!recordsOneChannel.show) continue;
+        let selectedData = recordsOneChannel.data[0];
+        for (const record of recordsOneChannel.data) {
+          selectedData =
+            Math.abs(selectedData.time - mouseFocus) <
+            Math.abs(record.time - mouseFocus)
+              ? selectedData
+              : record;
+        }
+        if (!selectedData) continue;
+
         const upperDate = timeParse(
           Math.floor(selectedData.time / 1000).toString()
         );
 
-        // Sets legend text.
-        recordsOneChannel.focusTime =
-          timeFormat(upperDate) + '.' + Math.floor(mouseFocus % 1000);
-        recordsOneChannel.focusDate = dateFormat(upperDate);
-        recordsOneChannel.focusPower = selectedData.value.toString();
-
         // Sets focus point.
         this.svgLine
           .select('.' + this.getChannelCircleClassName(recordsOneChannel.name))
+          .style('opacity', 1)
           .attr('cx', this.xScale(selectedData.time))
           .attr('cy', this.yScale(selectedData.value));
+
+        this.svg
+          .select('.' + this.getFocusTextClassName(recordsOneChannel.name))
+          .attr('opacity', 1)
+          .attr('x', this.xScale(selectedData.time))
+          .attr('y', this.yScale(selectedData.value) - this.chartPadding)
+          .text(selectedData.value.toString());
+
+        this.mouseDate = dateFormat(upperDate);
+        this.mouseTime =
+          timeFormat(upperDate) + '.' + Math.floor(mouseFocus % 1000);
+      }
+    };
+
+    const removeFocus = () => {
+      for (const recordsOneChannel of this.records) {
+        recordsOneChannel.focusTime = undefined;
```

Comment:
Set a property as undefined is a bad practice, try to use empty string instead

Reply:
Fixed by removing this property.